cagov/ui-claim-tracker
# Adds Timeout Functionality

### Changes
Adds the Timeout Modal component and integrates it into the Main component of the site. After 30 minutes the site times out.

Adds a TimedOut story to the Main component that displays the timed out state.

### Testing

Test behavior:
 - Checkout the branch
 - Edit `components/TimeoutModal.tsx` line 18-19 to be a two minute timeout with the modal popping up with 1 m to go
   ```
   const TIMEOUT_MS = 2 * 60 * 1000 
   const TIMEOUT_DISPLAY_TIME_IN_MINUTES = 1
   ```  
 - Load up the app and wait a minute
 - Observe the warning modal
 - Wait another minute
 - Confirm the site reloads to EDD
 - (bonus round: if you swap to Spanish the modal should display in Spanish)
 - discard your changes
 
### Screenshot
![Timed out view](https://user-images.githubusercontent.com/6129479/117824911-dbc78500-b23c-11eb-8d26-85ef17865670.png)

---

Resolves #114 

### Acceptance Criteria

 - [ ] A timeout modal on staging that matches the one from retroactive certification (ignoring styling for now) and redirects to the appropriate BPO URL
